### PR TITLE
Added support for FileClient and DirectoryClient ForceCloseHandles x-…

### PR DESCRIPTION
**Pull request by seanmcc-msft**
…ms-number-of-handles-failed header

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/CloseHandlesResult.cs [hunk 1]
```diff
@@ -15,6 +17,11 @@ public class CloseHandlesResult
         /// </summary>
         public int ClosedHandlesCount { get; internal set; }
 
+        /// <summary>
+        /// The number of file handles that fialed to close.
+        /// </summary>
+        public int FailedHandlesCount { get; internal set; }
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/Models/CloseHandlesResult.cs]: We can change this property name if One SDK doesn't like it.

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/CloseHandlesResult.cs [hunk 2]
```diff
@@ -31,10 +38,32 @@ public static partial class FileModelFactory
...
+                FailedHandlesCount = failedHandlesCount
+            };
+    }
+}
+
+namespace Azure.Storage.Files.Shares
+{
+    internal static partial class ShareExtensions
+    {
+        internal static CloseHandlesResult ToCloseHandlesResult(this Response<StorageClosedHandlesSegment> storageClosedHandlesSegment)
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/Models/CloseHandlesResult.cs]: Consolidated StorageClosedHandlesSegment -> CloseHandlesResult conversion here.

File: sdk/storage/Azure.Storage.Files.Shares/tests/ServiceClientTests.cs [hunk 3]
```diff
@@ -71,7 +71,7 @@ public async Task GetPropertiesAsync_Error()
             // Act
             await TestHelper.AssertExpectedExceptionAsync<RequestFailedException>(
                 service.GetPropertiesAsync(),
-                e => Assert.AreEqual("AuthenticationFailed", e.ErrorCode.Split('\n')[0]));
+                e => Assert.AreEqual("InvalidHeaderValue", e.ErrorCode.Split('\n')[0]));
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/tests/ServiceClientTests.cs]: This error changed in the 2019-07-07 service version.

**Comment by seanmcc-msft** [comment]
https://github.com/Azure/azure-sdk-for-net/pull/8989

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/CloseHandlesResult.cs [hunk 4]
```diff
@@ -15,6 +17,11 @@ public class CloseHandlesResult
         /// </summary>
         public int ClosedHandlesCount { get; internal set; }
 
+        /// <summary>
+        /// The number of file handles that fialed to close.
```
Comment (gapra-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/Models/CloseHandlesResult.cs]: typo in failed

**Pull request closed without merging**